tests/forte/processors/data_augment/algorithms/word_splitting_processor_test.py

```diff
@@ -0,0 +1,114 @@
+# limitations under the License.
+"""
+Unit tests for Random Word Splitting Data Augmentation processor
+"""
+
+import unittest
+import random
+
+from forte.data.selector import AllPackSelector
+from forte.pipeline import Pipeline
+from forte.data.multi_pack import MultiPack
+from forte.data.readers import StringReader
+from forte.data.caster import MultiPackBoxer
+from forte.processors.misc import PeriodSentenceSplitter
+from forte.processors.data_augment.algorithms.word_splitting_processor import (
+    RandomWordSplitDataAugmentProcessor,
+)
+
+from forte.processors.misc import WhiteSpaceTokenizer
+from ft.onto.base_ontology import Token
+
+from ddt import ddt, data, unpack
+
+
+@ddt
+class TestWordSplittingProcessor(unittest.TestCase):
+    def setUp(self):
+        random.seed(8)
+        self.nlp = Pipeline[MultiPack]()
+
+        boxer_config = {"pack_name": "input_src"}
+
+        self.nlp.set_reader(reader=StringReader())
+        self.nlp.add(PeriodSentenceSplitter())
+        self.nlp.add(component=MultiPackBoxer(), config=boxer_config)
+        self.nlp.add(
+            component=WhiteSpaceTokenizer(), selector=AllPackSelector()
+        )
+
+    @data(
+        (
+            [
+                "Mary and Samantha arrived at the bus station on time . "
+                "But they had to wait until noon for the bus ."
+            ],
+            [
+                "Mary and Samantha arrived at the bus statio n on time . "
+                "But t hey h ad to wait until noon for the bus ."
+            ],
+            [
+                [
+                    "Mary",
+                    "and",
+                    "Samantha",
+                    "arrived",
+                    "at",
+                    "the",
+                    "bus",
+                    "statio",
+                    " n",
+                    "on",
+                    "time",
+                    ".",
+                    "But",
+                    "t",
+                    " hey",
+                    "h",
+                    " ad",
+                    "to",
+                    "wait",
+                    "until",
+                    "noon",
+                    "for",
+                    "the",
+                    "bus",
+                    ".",
+                ]
+            ],
+            [["station", "they", "had"]],
+        )
+    )
+    @unpack
+    def test_word_splitting_processor(
+        self, texts, expected_outputs, expected_tokens, unnecessary_tokens
+    ):
+        self.nlp.add(component=RandomWordSplitDataAugmentProcessor())
+        self.nlp.initialize()
+
+        for idx, m_pack in enumerate(self.nlp.process_dataset(texts)):
+            aug_pack = m_pack.get_pack("augmented_input_src")
+
+            self.assertEqual(aug_pack.text, expected_outputs[idx])
+
+            for j, token in enumerate(aug_pack.get(Token)):
+                self.assertEqual(token.text, expected_tokens[idx][j])
```

Comment:
Can we add some test cases for other Annotation?

For example, let's label "Mary and Samantha" and "the bus station" as "EntityMention", and check if they can be recovered correctly?